source/1_engine/rex_engine/src/entrypoint.cpp

```diff
@@ -0,0 +1,10 @@
+#include "rex_engine/entrypoint.h"
+
+#include "rex_std/string.h"
+
+namespace rex
+{
+  void pre_app_entry(REX_MAYBE_UNUSED const char8* cmdLine) {}
+
+  void post_app_shutdown() {}
```

Comment:
I'm not a big fan of the pre and post design ... this always has to be executed in order and is easy to mess up IMO.

Reply:
Is post_app_shutdown even used?

Reply:
How is it easy to mess up? It's literally in the name when you have to use it?
And it's the engine that calls it so the client shouldn't worry about it

Reply:
This is the console app ...

Reply:
they're used with the file system

Reply:
Anything that requires an order in which something is called can be messed up. It feels iffy =/ don't add them in global space encapsulate them and have one function queue them

Reply:
it's well defined as its marked with "pre" and "post" so I don't agree with that it's easy to mess up.

Reply:
We had this structure in the past within the application with pre init and post init, I don't agree with how this is done. Encapsulate them within a class or a function an let the class call/function them in the right order. Just as I did for the context

Reply:
Why do you need them, what do we require before we enter the application?

Reply:
Or after we enter the application?

Reply:
vfs initialization.
I've told you this before

Reply:
And why can't we initialize this in CoreApplication?

Reply:
because by then it's too late.
the client might want to access files for its app creation params.
they can't do that if vfs is not initialized by that point yet

Reply:
But initialization of the VFS is our concern, the client has nothing to do with this right? So can't we make a function within entrypoint.cpp (and thus not expose it in the header ) that would initialize the VFS?

Reply:
Then a client can still use the VFS in app_entry